modules/account-lib/src/coin/cspr/delegateBuilder.ts

```diff
@@ -0,0 +1,146 @@
+import { BaseCoin as CoinConfig } from '@bitgo/statics/dist/src/base';
+import { CLValue, PublicKey } from 'casper-client-sdk';
+import { BuildTransactionError, InvalidParameterValueError, SigningError } from '../baseCoin/errors';
+import { BaseKey } from '../baseCoin/iface';
+import { TransactionType, StakingOperationTypes } from '../baseCoin';
+import { TransactionBuilder, DEFAULT_M } from './transactionBuilder';
+import { Transaction } from './transaction';
+import {
+  SECP256K1_PREFIX,
+  TRANSACTION_TYPE,
+  DELEGATE_VALIDATOR,
+  DELEGATE_FROM_ADDRESS,
+  STAKING_TYPE,
+} from './constants';
+import { isValidAmount, isValidAddress, getTransferAmount, getDelegatorAddress, getValidatorAddress } from './utils';
+
+export class DelegateBuilder extends TransactionBuilder {
+  private _validator: string;
+  private _delegator: string;
+  private _action: string;
+  private _amount: string;
+
+  /**
+   * Public constructor.
+   *
+   * @param {CoinConfig} _coinConfig Coin configuration object
+   */
+  constructor(_coinConfig: Readonly<CoinConfig>) {
+    super(_coinConfig);
+    this._action = 'delegate';
+  }
+
+  /** @inheritdoc */
+  protected async buildImplementation(): Promise<Transaction> {
+    const extraArguments = new Map<string, CLValue>();
+    if (this._delegator && this._validator) {
+      extraArguments.set(TRANSACTION_TYPE, CLValue.string(TransactionType[TransactionType.StakingLock]));
+      extraArguments.set(STAKING_TYPE, CLValue.string(StakingOperationTypes[StakingOperationTypes.LOCK]));
+      // TODO(STLX-1691): We are send the destination address as string until impediment STLX-1691 is fixed.
+      // After that we will change this to send an instance of PublicKey instead.
+      extraArguments.set(DELEGATE_FROM_ADDRESS, CLValue.string(this._delegator));
+      extraArguments.set(DELEGATE_VALIDATOR, CLValue.string(this._validator));
+    }
+    this._session = {
+      action: this._action,
+      delegator: PublicKey.fromHex(SECP256K1_PREFIX + this._delegator),
+      validator: PublicKey.fromHex(SECP256K1_PREFIX + this._validator),
+      amount: this._amount,
+      extraArguments: extraArguments,
+    };
```

Comment:
I think this should be done like in WalletInitializationBuilder.
Since delgate operation will be done using a contract, we should use ModuleBytes to send the contract and pass contract parameters as RuntimeArgs.

https://github.com/BitGo/BitGoJS/blob/master/modules/account-lib/src/coin/cspr/walletInitializationBuilder.ts#L24-L36